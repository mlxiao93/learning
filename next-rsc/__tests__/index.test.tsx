import ForTesting from '@/components/for-testing';
import { render, fireEvent, act } from '@testing-library/react';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<ForTesting />).toJSON();
  expect(tree).toMatchSnapshot();
})

it('increments the count when button is clicked', async () => {
  const { getByText } = render(<ForTesting />);
  const button = getByText('count++');
  await act(() => {
    fireEvent.click(button);
  })
  expect(getByText('1')).toBeInTheDocument();
});

// TODO use act