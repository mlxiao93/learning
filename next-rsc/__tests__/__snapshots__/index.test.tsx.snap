// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  >
    count++
  </button>
</div>
`;
